docs/spec/concepts.rst

```diff
@@ -3,97 +3,186 @@
 Type system concepts
 ====================
 
-.. _`union-types`:
+Static, dynamic, and gradual typing
+-----------------------------------
+
+A **statically-typed** programming language runs a type-checker before running
+a program. The program is required to be well-typed according to the language's
+type system. The type system assigns a type to all expressions in the language
+and verifies that their uses obey the typing rules. Normally, a program that is
+not well-typed (i.e., one that contains a type error) will not run. Java and
+C++ are examples of statically-typed object-oriented languages.
+
+A **dynamically-typed** programming language does not run a type-checker before
+running a program. Instead, it will check the types of values before performing
+operations on them at runtime. This is not to say that the language is
+"untyped". Values at runtime have a type and their uses obey typing rules. Not
+every operation will be checked, but certain primitive operations in the
+language such as attribute access or arithmetic will be. Python was
+historically a dynamically-typed language.
+
+**Gradual typing** is the name for a specific way to combine static and dynamic
+typing. It allows mixing static and dynamic checking at a fine level of
```

Comment:
It can be a coarse granularity too. We have macro gradual typing which enables us to type at the module level (such as Typed Racket) and micro-gradual typing which allows us to type at the expression level (such as Reticulated Python).

Reply:
Good point. Do you think this is important to clarify here? I guess it is still true that gradual typing _allows_ mixing at fine granularity.

Reply:
Oh that's true! 
If we're not talking about typed/untyped boundaries, runtime type checks or performance, then it may not be important to mention.

Reply:
We shouldn't say things that are wrong, and I think it's a good point that this isn't quite right.  Probably just better to say "Python allows" instead of "It allows" and they we don't need to mention Python again in the next sentence.